### PR TITLE
make ConnectivityMonitorFactory injectable

### PR DESCRIPTION
**Pull request by emakar**
Our app needs some higher level connectivity states not related to just android.net.ConnectivityManager (e.g. work via wi-fi ony or offline at despite device connectivity), so this change allows Glide to respect such states

### CODE REVIEW

**Comment by emakar** [comment]
I signed it!

**Comment by TWiStErRob** [comment]
Hi, here's what I would change:
 1. I think the `connectivity` package is unnecessary, it just fragments the packages unnecessarily, also not introducing it keeps more compatibility
 2. `ConnectivityMonitorFactoryImpl` -> `DefaultConnectivityMonitorFactory` as it tries to create `DefaultConnectivityMonitor`. (AFIAR the Impl suffix is not used anywhere else in Glide)
 3. The FQCN in JavaDoc links is the style Sam's going for I think.

**Comment by emakar** [comment]
Thanks! Committed code style and naming fixes. I've rebased the changes as well

**Comment by TWiStErRob** [comment]
Is that new constructor in `RequestManager` used anywhere?

**Comment by emakar** [comment]
Nope, just telescoping constructor in case someone decides to use it direclty (so as for now it's used in another ctor only)

**Comment by TWiStErRob** [comment]
I don't think you ever want to use `RequestManager` constructor directly, as far as I know it's public because it's accessed from another package. It should be always created/returned by a `Glide.with` call.

**Comment by emakar** [comment]
Agreed, fixed

**Comment by emakar** [comment]
Rebased.

File: library/src/main/java/com/bumptech/glide/Glide.java [hunk 1]
```diff
@@ -290,6 +294,10 @@ public ArrayPool getArrayPool() {
     return arrayPool;
   }
 
+  public ConnectivityMonitorFactory getConnectivityMonitorFactory() {
```
Comment (sjudd) [inline library/src/main/java/com/bumptech/glide/Glide.java]: Can you make this package private?

File: library/src/test/java/com/bumptech/glide/manager/ConnectivityMonitorFactoryTest.java [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
   @Before
```
Comment (sjudd) [inline library/src/test/java/com/bumptech/glide/manager/ConnectivityMonitorFactoryTest.java]: Can you rename this DefaultConnectivityMonitorFactoryTest to match the class name?

**Comment by sjudd** [comment]
Other than two small nits, looks good to me. Thanks for putting this up.

If any of the changes you need for your app are generally applicable, I'd welcome a pull request to make them to the default connectivity monitor as well.

**Comment by emakar** [comment]
Thanks! Fixed encapsulation and naming.
As for the custom monitor I think it's rather specific than general. Our main purpose is just notifying ConnectivityListener. And it's performed with complicated "connectivity monitor" which besides common implementation uses several app-private restrictions (bunch of network modes, etc). So such ConnectivityMonitor is just a thin client/observer of existent complicated api.

**Comment by sjudd** [comment]
Merged here: b334f4f145a899d3e2c013d222d4ebcf6eb94f0d

Sorry for the lack of credit, we're still trying to work out the kinks in this new merging scheme...

**Pull request closed without merging**